### PR TITLE
Switch SVG path parsing to fonttools

### LINKED ISSUE #525: fonttools.svgPath.path.parse_path() to replace svg.path?
I just discovered that fonttools have a module "svgLib", containing a class [svgLib.path](https://fonttools.readthedocs.io/en/latest/svgLib/path/index.html), which again has a method `parse_path()`. 

I wonder if this could be used to replace the "svg.path" package that we currently use, reducing the number of dependencies for fpdf2 again.

### PR DESCRIPTION
**Pull request by gmischler**
e.g. Fixes #525

**Checklist**:

- [x] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.
- [x] A unit test is covering the code added / modified by this PR
- [x] This PR is ready to be merged 
- [NA] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder
- [x] A mention of the change is present in `CHANGELOG.md`

The nice part is that the new parser resolves everything to absolute coordinates, and it does so in a very simple and elegant way. For us this means that many class definitions dealing with relative movements have become redundant. After removing them, sgv.py is now significantly shorter.

At first, I couldn't get the "exponentated numbers" to parse correctly. Then I found that my copy of fonttools was slightly outdated, and they had just recently fixed a bug quite similar to our own #376. I have now pinned fonttools to the current release 4.37.2 or newer as well to make sure we have this fix available.

Previously, the "comma seperation" test converted "M0,1l2,3" into `[M(0, 1), L2, 3)]`. I don't think this was correct, since a relative line_to should end up at absolute (2,4). I didn't analyze if this was our own bug or one in "svg.path", but now it's gone anyway.

When a path gets closed with "Z/z", the fonttools parser adds an explicit extra "L" back to the start point. They may need this internally, but it has no purpose for us. I had to add methods to `drawing._graphics_context()` and `drawing.PaintedPath()` for removing the last element of a path again to eliminate this.

I had to replace a few of the reference PDFs. They all looked identical visually, but the new ones were consistently a bit smaller.

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/PyFPDF/fpdf2/blob/master/LICENSE).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ This can also be enabled programmatically with `warnings.simplefilter('default',
 
 ## [2.5.7] - 2022-09-08
 ### Changed
+- Since the fonttools library offers similar functionality, the dependency to "svg.path" is gone again, thanks to @gmischler; [#525](https://github.com/PyFPDF/fpdf2/issues/525)
```
Comment (Lucas-C) [inline CHANGELOG.md:29]: 👍

File: docs/index.md [hunk 2]
```diff
@@ -40,7 +40,7 @@ Go try it **now** online in a Jupyter notebook: [![Open In Colab](https://colab.
 * [Annotations](Annotations.md), including text highlights, and [file attachments](FileAttachments.md)
 * [Presentation mode](Presentations.md) with control over page display duration & transitions
 * Optional basic Markdown-like styling: `**bold**, __italics__`
-* It has very few dependencies: [Pillow](https://pillow.readthedocs.io/en/stable/), [defusedxml](https://pypi.org/project/defusedxml/), [svg.path](https://pypi.org/project/svg.path/) & [fonttools](https://pypi.org/project/fonttools/)
+* It has very few dependencies: [Pillow](https://pillow.readthedocs.io/en/stable/), [defusedxml](https://pypi.org/project/defusedxml/), & [fonttools](https://pypi.org/project/fonttools/)
 * Can render [mathematical equations & charts](https://pyfpdf.github.io/fpdf2/Maths.html)
 * Many example scripts available throughout this documentation, including usage examples with [Django](https://www.djangoproject.com/), [Flask](https://flask.palletsprojects.com), [streamlit](https://streamlit.io/), AWS lambdas... : [Usage in web APIs](UsageInWebAPI.md)
 * Unit tests with `qpdf`-based PDF diffing, and PDF samples validation using 3 different checkers:
```
Comment (Lucas-C) [inline docs/index.md:46]: Are the comment & code on lines 77-79 below still meaningful after introducing the dependency on `fonttools`?
Else we should probably remove them.

File: fpdf/svg.py [hunk 3]
```diff
@@ -566,293 +548,74 @@ def convert_transforms(tfstr):
...
-
-    def render(self, path_gsds, style, last_item, initial_point):
-        if isinstance(last_item, QuadraticBezierCurve):
-            ctrl = (2 * last_item.end) - last_item.ctrl
-        else:
-            ctrl = last_item.end_point
-
-        return QuadraticBezierCurve(ctrl, self.end).render(
-            path_gsds, style, last_item, initial_point
+class PathPen:
```
Comment (Lucas-C) [inline fpdf/svg.py:551]: Why not inherinting from `fontTools.pens.basePen.AbstractPen`?
_cf._ https://fonttools.readthedocs.io/en/latest/pens/basePen.html?fontTools.pens.basePen.AbstractPen#fontTools.pens.basePen.AbstractPen

If there is a reason not to inherit from this class, could you add a mention and a link to the pen protocol in this class docstring please? _cf._ https://fonttools.readthedocs.io/en/latest/pens/basePen.html

File: test/drawing/test_drawing.py [hunk 4]
```diff
@@ -489,6 +489,22 @@ def test_copy(self):
         assert style.fill_opacity != copied.fill_opacity
 
 
+def test_path_start_relative():
+    """
+    Test starting a path with a relative move.
+    The parametrized series below can't cover that case.
```
Comment (Lucas-C) [inline test/drawing/test_drawing.py:495]: 👍

**Review by Lucas-C — APPROVED** [review]
Good job!
Nice code shrinking here, thank you @gmischler!

By the way, don't hesitate to ping by email whenever you want to get the role of a maintainer on this project, I still think `fpdf2` would benefit from it and you clearly deserve it

File: docs/index.md (hunk 2 above)
Comment (gmischler) [reply docs/index.md:46]: Theoretically, we could enable running without fonttools, which would disable all text related functions.
But that is probably not a very common use case...

I'll remove that section.

File: fpdf/svg.py (hunk 3 above)
Comment (gmischler) [reply fpdf/svg.py:551]: I didn't even see that one...

I guess it's mostly a formality, but definitively good form to use the appropriate ABS.

**Comment by Lucas-C** [comment]
```
$ pylint fpdf test tutorial/tuto*.py
************* Module fpdf.svg
fpdf/svg.py:552:0: W0223: Method 'addComponent' is abstract in class 'AbstractPen' but is not overridden (abstract-method)
fpdf/svg.py:558:4: W0237: Parameter 'pt' has been renamed to 'end' in overridden 'PathPen.moveTo' method (arguments-renamed)
fpdf/svg.py:564:4: W0237: Parameter 'pt' has been renamed to 'end' in overridden 'PathPen.lineTo' method (arguments-renamed)
fpdf/svg.py:570:4: W0221: Number of parameters was 2 in 'AbstractPen.curveTo' and is now 4 in overridden 'PathPen.curveTo' method (arguments-differ)
fpdf/svg.py:570:4: W0221: Variadics removed in overridden 'PathPen.curveTo' method (arguments-differ)
fpdf/svg.py:583:4: W0221: Number of parameters was 2 in 'AbstractPen.qCurveTo' and is now 3 in overridden 'PathPen.qCurveTo' method (arguments-differ)
fpdf/svg.py:583:4: W0221: Variadics removed in overridden 'PathPen.qCurveTo' method (arguments-differ)
```

**Comment by Lucas-C** [comment]
To me, this looks ready to be merged 😊

File: setup.py [hunk 5]
```diff
@@ -31,9 +31,8 @@
         package_dir={"fpdf": "fpdf"},
         install_requires=[
             "defusedxml",
-            "svg.path",
             "Pillow>=6.2.2",  # minimum version tested there: https://github.com/PyFPDF/fpdf2/actions/runs/2295868575
-            "fonttools",
+            "fonttools>=4.37.2",
```
Comment (Lucas-C) [inline setup.py:35]: The current minium version is `4.30` on `master`: is this really the minimum version now @gmischler?

Comment (gmischler) [reply setup.py:35]: The absolute minimum is [4.34.0](https://github.com/fonttools/fonttools/releases/tag/4.34.0), which is when the exponentials were fixed. I just picked the most current one at the time, but I don't really care as long as it works.

Comment (Lucas-C) [reply setup.py:35]: OK then could you change this to 4.34.0 please?

**Pull request merged**